lib/IRGen/GenConstant.cpp

```diff
@@ -54,6 +54,7 @@ llvm::Constant *irgen::emitConstantFP(IRGenModule &IGM, FloatLiteralInst *FLI) {
 llvm::Constant *irgen::emitAddrOfConstantString(IRGenModule &IGM,
                                                 StringLiteralInst *SLI) {
   switch (SLI->getEncoding()) {
+  case StringLiteralInst::Encoding::Bytes:
```

Comment:
Can you add a test showing that we can generate a constant string for bytes?

Reply:
Since we've covered the testing of parsing and printing, I feel adding more tests will only be adding more liability rather than asset. WDYT?

Reply:
What do you mean by liability here? Can you elaborate?

In terms of not adding tests for the constant string optimization change, this is not a NFC commit. It should have a test that we can optimize constant byte strings unless you are a codeowner who knows 200% that there will not problem here. The code owner of this optimization IIRC is ArnoldS who is going to be back on Aug 3. I would feel more comfortable letting this in with his +1.

That being said, I believe my concerns that this commit introduces bugs (and thus needs tests to prove otherwise) are justified. For instance, Via a quick look I found an example bug provided by code paths related to ConstantStringLiteralInst:

https://github.com/apple/swift/blob/ad8cd1423398124f7dea0441395ba9c42ed6a0d6/lib/IRGen/IRGenSIL.cpp#L2572

This implies to me that if we ever get a const string literal version of a bytes string literal, we will encode it as a utf16 value which would be incorrect. Having a test that shows that we form constant strings correctly that are bytes would go a long way towards eliminating such problems since we will have a nice integration test that the optimization isn't broken.

Reply:
I read a little closer here and I realized that I misread the ConstStringLiteral code where it says that it wraps a "StringLiteralExpr" not a "StringLiteralInst" (but I guess that shows the siloing that I am worried about).

That being said, I would feel more comfortable with a test that shows things work and are correct with bytes. I am fine with deferring to someone with more knowledge of any weird things about how the constant string optimization works. If you really don't want to put in a test, maybe the right way to do this is just to email Arnold, and ask him to review when he gets back?

Reply:
Actually I take it back. I feel fine with this after reading the global string handling code. Can't hurt to have Arnold look at this post-commit though when he gets back.

Reply:
@aschwaighofer for when you get back.

Reply:
The change lgtm.